Make {{ devices }} list optional

eduardegorov opened the pull request:
Define it as empty list by default, remove unneccessary 'default([])' checks

Signed-off-by: Eduard Egorov eduard.egorov@icl-services.com

leseb commented:
Thanks for this PR, can you also do the same for `raw_journal_devices`? Thanks!

eduardegorov commented:
Sure, I will. Thanks!

leseb commented:
@eduardegorov any updates?

eduardegorov commented:
@leseb , sorry! If it is still actual, please pull.

Note: this variable is also defined in `roles/ceph-common/defaults/main.yml`:

```yml
##################
# Temporary Vars #
##################
# NOTE(SamYaple): These vars are set here to they are defined before use. They
# should be removed after a refactor has properly seperated all the checks into
# the appropriate roles.

journal_collocation: False
raw_multi_journal: False
osd_directory: False
bluestore: False
dmcrypt_journal_collocation: False
dmcrypt_dedicated_journal: False
raw_journal_devices: {}                       <--- HERE

osd_auto_discovery: False
```

The problem is I don't know what is the difference between {} and [] but looks like it's the same.. Let me know if any updates are needed.

leseb commented:
Hey {} is a dictionary where [] is an array

leseb commented:
test this please

leseb commented:
One last thing, please update `roles/ceph-common/defaults/main.yml` and run `generate_group_vars_sample.sh`. Thanks!

eduardegorov commented:
Fixed, thank you, Sebastien! I had already updated `roles/ceph-common/defaults/main.yml` and run `generate_group_vars_sample.sh` in the 3rd commit - am I missing something there?

leseb commented:
Hum I'm just not sure why length > 1 didn't work and == 0 seems to work.

leseb commented on roles/ceph-common/tasks/checks/check_mandatory_vars.yml:
```diff
@@ -115,8 +115,8 @@
-    - raw_journal_devices is not defined
-      or devices is not defined
+    - raw_journal_devices|length == 0
+      or devices|length == 0
```
LGTM, can we just rephrase the fail sentence for clarity with something like: `please provide devices and raw journal devices to your osd scenario`?

Thanks!

eduardegorov replied:
Of course, done, thank you for your patience! Let me know if anything else is needed, I'm a complete newbie here

leseb replied:
Perfect thanks, I think we are good to go with this one :)

leseb commented on roles/ceph-common/defaults/main.yml:
```diff
@@ -360,7 +360,7 @@ osd_directory: False
 dmcrypt_journal_collocation: False
 dmcrypt_dedicated_journal: False
-raw_journal_devices: {}
+raw_journal_devices: []
```
Thanks for the intensive testing, we are almost there :).

leseb commented:
test this please

leseb commented:
test this please

leseb commented:
test this please

leseb commented:
test this please
